graphql/schemabuilder/pagination.go

```diff
@@ -281,9 +299,19 @@ func getConnection(key string, nodes []interface{}, args ConnectionArgs) (Connec
 		startCursor = edges[0].Cursor
 	}
 
+	if returnsConnInfo {
+		connInfo := out[1].Interface().(PaginationInfo)
+		pageInfo := PageInfo{
+			HasNextPage: connInfo.HasNextPage,
+			HasPrevPage: connInfo.HasPrevPage,
+			StartCursor: startCursor,
+			EndCursor:   endCursor,
+		}
+		return Connection{TotalCount: connInfo.TotalCount(), Edges: edges, PageInfo: pageInfo}, nil
```

Comment:
think we should invoke `TotalCount` only if the client requests it. i.e. it should be registered as a field func on `PageInfo`, rather than always invoking it.

Reply:
you mean registered as a field func on connection?